### PR TITLE
Upgrade Jakarta Mail to latest spec and implementation.

### PR DESCRIPTION
**Pull request by wetted**
- Upgrade Jakarta mail from 2.0.1 to 2.1.2. This includes changes to the dependency coordinates, which separates api and implementation.
- Add Eclipse Angus dependency, which is the direct successor of JakartaMail implementation.
- Fix links to point to moved Jakarta Mail Specification website, and update docs.

### CODE REVIEW

**Comment by wetted** [comment]
@melix I'm not sure I handled the addition of the implementation dependency correctly, in the catalog and build files. Please check me on this.

**Review by sdelamo — CHANGES_REQUESTED** [review]
It misses a mention in the what's new or breaking changes about the move to angus

**Review by melix — APPROVED** [review]

File: email-javamail-composer/build.gradle.kts [hunk 1]
```diff
@@ -7,6 +7,8 @@ dependencies {
     api(projects.micronautEmail)
     api(libs.managed.jakarta.mail)
     implementation(mnValidation.micronaut.validation)
+    runtimeOnly(libs.managed.eclipse.angus)
```
Comment (sdelamo) [inline email-javamail-composer/build.gradle.kts:10]: I think we should move this dependency to `testRuntimeOnly` and tell the user in the documentation they have to add the implementation they choose as a runtimeOnly dependency. 

We could say to you java email you have to add: 

implementation("io.micronaut.email:micronaut-email-javamail")
and: 
runtimeOnly(libs.managed.eclipse.angus)

**Review by sdelamo — CHANGES_REQUESTED** [review]
move   `libs.managed.eclipse.angus` to testRuntimeOnly.

**Review by sdelamo — APPROVED** [review]

**Pull request merged**